src/overlays/actors/ovl_Bg_Po_Syokudai/z_bg_po_syokudai.c

```diff
@@ -21,11 +46,128 @@ const ActorInit Bg_Po_Syokudai_InitVars = {
     (ActorFunc)BgPoSyokudai_Update,
     (ActorFunc)BgPoSyokudai_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Syokudai/BgPoSyokudai_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Syokudai/BgPoSyokudai_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern Gfx D_060003A0[]; // Display List
+extern Gfx D_0404D4E0[]; // Display List
+
+void BgPoSyokudai_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    void* pad;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+
+    this->unk_14C = (THIS->actor.params >> 0x8) & 0xFF;
+    this->actor.params &= 0x3F;
+
+    this->actor.colChkInfo.mass = 0xFF;
+
+    this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+    Lights_PointGlowSetInfo(&this->lightInfo, (s16)this->actor.posRot.pos.x, ((s16)this->actor.posRot.pos.y) + 0x41,
+                            (s16)this->actor.posRot.pos.z, 0, 0, 0, 0);
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+
+    this->collider.dim.pos.x = (s16)this->actor.posRot.pos.x;
+    this->collider.dim.pos.y = (s16)this->actor.posRot.pos.y;
+    this->collider.dim.pos.z = (s16)this->actor.posRot.pos.z;
+
+    if (!this->unk_14C && Flags_GetSwitch(globalCtx, 0x1F) && Flags_GetSwitch(globalCtx, 0x1E) &&
```

Comment:
I would prefer `this->unk_14C != 0` because `unk_14C` is clearly not a boolean value (0/1)
for example it indexes into `primColors` which is an array of length 4